File: 4-3-digit-primes-starting-with-5/primes.py
def is_prime(n):
    """Return True if n is a prime number."""
    if n < 2:
        return False
    if n == 2:
        return True
    if n % 2 == 0:
        return False
    i = 3
    while i * i <= n:
        if n % i == 0:
            return False
        i += 2
    return True


def three_digit_primes_starting_with_five():
    """Return all 3-digit prime numbers whose first digit is 5 (i.e. 500-599)."""
    return [n for n in range(500, 600) if is_prime(n)]


if __name__ == "__main__":
    primes = three_digit_primes_starting_with_five()
    print(f"Count: {len(primes)}")
    print("Primes:")
    for p in primes:
        print(p)

File: 4-3-digit-primes-starting-with-5/primes.c
#include <stdio.h>
#include <stdbool.h>

static bool is_prime(int n) {
    if (n < 2)        return false;
    if (n == 2)       return true;
    if (n % 2 == 0)   return false;

    for (int i = 3; i * i <= n; i += 2) {
        if (n % i == 0) {
            return false;
        }
    }
    return true;
}

int main(void) {
    int count = 0;
    int primes[100];

    for (int n = 500; n <= 599; ++n) {
        if (is_prime(n)) {
            primes[count++] = n;
        }
    }

    printf("Count: %d\n", count);
    printf("Primes:\n");
    for (int i = 0; i < count; ++i) {
        printf("%d\n", primes[i]);
    }
    return 0;
}
